server/message_hooks.go

```diff
@@ -69,106 +74,127 @@ func (p *Plugin) MessageHasBeenPosted(_ *plugin.Context, post *model.Post) {
 				p.API.LogWarn("Unable to handle message sent", "error", err.Error())
 			}
 		}
-		return
+		return nil
 	}
 
 	user, _ := p.API.GetUser(post.UserId)
 
 	_, err = p.Send(link.MSTeamsTeam, link.MSTeamsChannel, user, post)
 	if err != nil {
 		p.API.LogWarn("Unable to handle message sent", "error", err.Error())
+		return err
 	}
+	return nil
 }
 
 func (p *Plugin) ReactionHasBeenAdded(c *plugin.Context, reaction *model.Reaction) {
+	p.GetMetrics().ObserveMessageHooksEvent("reaction_added")
 	updateRequired := true
 	if c.RequestId == "" {
 		_, ignoreHookForReaction := p.activityHandler.IgnorePluginHooksMap.LoadAndDelete(fmt.Sprintf("%s_%s_%s", reaction.PostId, reaction.UserId, reaction.EmojiName))
 		updateRequired = !ignoreHookForReaction
 	}
+	_ = p.syncReactionAdded(reaction, updateRequired)
```

Comment:
See above for proposal on handling this error. (I won't comment everywhere this applies.)